File: frontend/src/types/hazard.ts
export type HazardCategory =
  | 'physical'
  | 'chemical'
  | 'biological'
  | 'ergonomic'
  | 'electrical'
  | 'environmental';

export type RiskLevel = 'critical' | 'high' | 'medium' | 'low';

export interface Hazard {
  id: string;
  category: HazardCategory;
  name: string;
  description: string;
  risk_level: RiskLevel;
  location?: string;
  potential_consequences: string[];
  preventive_measures: string[];
  legal_reference?: string;
}

export const categoryLabels: Record<HazardCategory, string> = {
  physical: '물리적 위험',
  chemical: '화학적 위험',
  biological: '생물학적 위험',
  ergonomic: '인간공학적 위험',
  electrical: '전기적 위험',
  environmental: '환경적 위험',
};

export const riskLevelLabels: Record<RiskLevel, string> = {
  critical: '즉시 조치 필요',
  high: '높음',
  medium: '중간',
  low: '낮음',
};

export const riskLevelColors: Record<RiskLevel, string> = {
  critical: 'risk-critical',
  high: 'risk-high',
  medium: 'risk-medium',
  low: 'risk-low',
};

File: backend/app/models/hazard.py
from pydantic import BaseModel
from enum import Enum
from typing import List, Optional


class HazardCategory(str, Enum):
    PHYSICAL = "physical"
    CHEMICAL = "chemical"
    BIOLOGICAL = "biological"
    ERGONOMIC = "ergonomic"
    ELECTRICAL = "electrical"
    ENVIRONMENTAL = "environmental"


class RiskLevel(str, Enum):
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class NormSummary(BaseModel):
    """위험요소에 연결된 규범명제 요약"""
    article_number: str
    legal_effect: str
    action: Optional[str] = None
    full_text: str


class FacetedHazardCodes(BaseModel):
    """Faceted 3축 canonical hazard codes (결정론적)"""
    accident_types: List[str] = []
    hazardous_agents: List[str] = []
    work_contexts: List[str] = []
    applied_rules: List[str] = []
    confidence: float = 0.0


class GptFreeObservation(BaseModel):
    """Track A: GPT 자유 관찰"""
    label: str
    description: str
    confidence: float
    visual_evidence: Optional[str] = None
    severity: str = "MEDIUM"


class CodeGapWarning(BaseModel):
    """코드 체계 gap 경고"""
    gap_type: str  # UNMAPPED / FORCED_FIT
    gpt_free_label: Optional[str] = None
    description: str


class PenaltyInfo(BaseModel):
    """벌칙 경로 정보"""
    article_code: str
    title: str
    criminal_employer_penalty: Optional[str] = None
    criminal_death_penalty: Optional[str] = None
    admin_max_fine: Optional[str] = None


class PenaltyCandidate(BaseModel):
    """Condition-aware penalty candidate from pen:PenaltyRule."""
    penalty_rule_id: str
    exposure_type: str  # direct_candidate | conditional
    condition_label: str
    subject_role: Optional[str] = None
    accident_outcome: Optional[str] = None
    violated_norm_id: Optional[str] = None
    violated_article_id: Optional[str] = None
    delegated_from_article_id: Optional[str] = None
    penalty_article_id: Optional[str] = None
    sanction_type: Optional[str] = None
    penalty_description: Optional[str] = None
    severity_score: Optional[int] = None
    basis_text: Optional[str] = None
    source_sr_id: Optional[str] = None


class PenaltyPath(BaseModel):
    """Business-facing penalty notice grouped into coarse outcome paths."""
    path_type: str  # general_incident | death | serious_accident
    title: str
    notice_level: str  # photo_based | external_fact_required | conditional
    summary: str
    penalty_rule_ids: List[str] = []
    penalty_descriptions: List[str] = []
    article_refs: List[dict] = []
    max_severity_score: Optional[int] = None
    source_sr_ids: List[str] = []


class Hazard(BaseModel):
    id: str
    category: HazardCategory
    name: str
    description: str
    risk_level: RiskLevel
    location: Optional[str] = None
    potential_consequences: List[str]
    preventive_measures: List[str]
    legal_reference: Optional[str] = None
    related_norms: List[NormSummary] = []
